client/src/store/post/actions.ts

```diff
@@ -0,0 +1,71 @@
+import { AuthAPI } from '../api';
+import { showAlert } from '../alert/actions';
+import { Dispatch } from 'redux';
+import {
+    ALL_POSTS_LOADED, CLEAR_LOADED,
+    GET_MORE_POSTS_SUCCESS,
+    GET_POSTS_PENDING,
+    GET_POSTS_SUCCESS,
+} from './actionTypes';
+import { store } from '../../App';
+import { IPost } from './reducers';
+
+export const getPostsPending = (): { type: string, loading: boolean } => ({
+    type: GET_POSTS_PENDING,
+    loading: true,
+});
+
+export const getPostsSuccess = (posts: IPost): { type: string, payload: any, loading: boolean } => ({
+    type: GET_POSTS_SUCCESS,
+    payload: posts,
+    loading: false,
+});
+
+export const getMorePostsSuccess = (posts: any): { type: string, payload: any, loading: boolean } => ({
+    type: GET_MORE_POSTS_SUCCESS,
+    payload: posts,
+    loading: false,
+});
+
+export const allPostsLoaded = (): { type: string, payload: boolean } => ({
+    type: ALL_POSTS_LOADED,
+    payload: true,
```

Comment:
Do we really need it here? Just set true in redux function

Reply:
Fixed

Reply:
I still see payload: true, set it in redux as it cant change